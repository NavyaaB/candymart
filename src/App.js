import React, { useState } from "react";
import "./styles/styles.scss";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import Category from "./components/Category";
import Navbar from "./components/Navbar";
import Product from "./components/Product";
import Footer from "./components/Footer";
import Cart from "./components/Cart";

export default function App() {
  const [cart, setAddToCart] = useState([]);

  return (
    <div>
      <BrowserRouter>
        <Navbar />
        <Routes>
          <Route path='/' element={<Category />} />
          <Route
            path='/category/:category_type'
            element={<Product cart={cart} setAddToCart={setAddToCart} />}
          />
          <Route
            path='/cart'
            element={<Cart cart={cart} setAddToCart={setAddToCart} />}
          />
        </Routes>
      </BrowserRouter>
      <Footer />
    </div>
  );
}
