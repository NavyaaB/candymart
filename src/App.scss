@use '_common';

// variables
$primary-color: 'Green';
$font-stack: Helvetica, sans-serif;
$secondary-color: '#f08080';
$font-size: '16px';

//nesting
//sass has has a clear nested and visual 
//hierarchy. CSS, on the other hand, doesn't.
nav {
  ul {
    margin: 0;
    padding: 0;
    list-style: none;
    //font-size: _common.$font-size;
  }

  li {
    display: inline-block;
  }

  a {
    display: block;
    padding: 6px 12px;
    text-decoration: none;
  }
}

//modules: @use filename

//mixins
@mixin theme($theme: DarkGray) {
  background: $theme;
  box-shadow: 0 0 1px rgba($theme, .25);
  color: #fff;
}

.info {
  @include theme;
}

.alert {
  @include theme($theme: DarkRed);
}

.success {
  @include theme($theme: DarkGreen);
}

.App {
  color: $primary-color;
  font-family: $font-stack;
}

.list {
  color: $secondary-color;
  text-decoration: none;
  font-size: $font-size;
}