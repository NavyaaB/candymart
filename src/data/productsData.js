const productsData = [
  { id: 1, price: 20, name: 'cake' },
  { id: 11, price: 20, name: 'cake' },
  { id: 122, price: 20, name: 'cake' },
  { id: 21, price: 20, name: 'cake' },
  { id: 13, price: 20, name: 'cake' },
  { id: 14, price: 20, name: 'cake' },
  { id: 15, price: 20, name: 'cake' },
  { id: 12, price: 20, name: 'cake' },
  { id: 114, price: 20, name: 'cake' },
  { id: 175, price: 20, name: 'cake' },
  { id: 16, price: 20, name: 'cake' },
  { id: 121, price: 20, name: 'cake' },
  { id: 155, price: 20, name: 'cake' },
  { id: 165, price: 20, name: 'cake' },
  { id: 19, price: 20, name: 'cake' },
];
export default productsData;
