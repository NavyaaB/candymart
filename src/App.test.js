import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import productsData from "./data/productsData";
import App from "./App";

const numOfProducts = productsData.length;
const nameOfProduct = productsData[0].name;

test("App snapshot test", () => {
  const component = renderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
//1
test("renders learn react link", () => {
  render(<App />);
  const linkElement = screen.getByText(/Shop By Category/i);
  expect(linkElement).toBeInTheDocument();
});
//2 number

test("Number of items = 15", () => {
  expect(numOfProducts).toBe(15);
});
//3 number
test("Number of items to be greater than or equal to 12", () => {
  expect(numOfProducts).toBeGreaterThanOrEqual(12);
});
//4 string
test("There is a ca in product name", () => {
  expect(nameOfProduct).toMatch(/ca/);
});
//5
test("The product name should contain cake", () => {
  expect(nameOfProduct).toContain("cake");
});
//6 arrays

describe("array tests", () => {
  const data2 = ["cake"];
  test("The list of products contains cake", () => {
    expect(["cake"]).toEqual(expect.arrayContaining(data2));
  });
});
//7 objects-ckeck the key
describe("object tests", () => {
  test("The list of products to have a property name", () => {
    expect(productsData[0]).toHaveProperty("name");
  });
  //8 objects - key and value
  test("The first product to have a property price and value of 20", () => {
    expect(productsData[0]).toHaveProperty("price", 20);
  });
});
