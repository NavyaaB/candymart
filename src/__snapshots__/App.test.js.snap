// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<div>
  <div
    className="navbar"
  >
    <ul>
      <li>
        <a
          href="#home"
        >
          My Store
        </a>
      </li>
      <li>
        <a
          href="#about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="#contact"
        >
          Contact
        </a>
      </li>
      <li
        style={
          Object {
            "float": "right",
          }
        }
      >
        <a
          href="/cart"
          onClick={[Function]}
        >
          <img
            alt="cart_image"
            className="cart_button active"
            src="https://www.phoca.cz/images/projects/phoca-cart-r.svg"
          />
        </a>
      </li>
    </ul>
  </div>
  <div>
    <div
      className="slider"
    >
      <p
        className="title"
      >
        Fun Fact!
      </p>
      <br />
      <p
        className="text"
      >
        Chocolate comes from a fruit tree; it’s made from a seed.
      </p>
    </div>
    <div
      className="content flexLayout"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        Restart
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        Prev
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        Next
      </button>
    </div>
  </div>
  <div
    className="header flexLayout"
  >
    <h2
      className="app_name"
    >
      Shop By Category
    </h2>
  </div>
  <div
    className="store_container"
  >
    <div
      className="product_wrapper"
      id={1}
      onClick={[Function]}
    >
      <img
        alt="category"
        className="category_image"
        height="150"
        src="https://storcpdkenticomedia.blob.core.windows.net/media/recipemanagementsystem/media/recipe-media-files/recipes/retail/x17/16714-birthday-cake-600x600.jpg?ext=.jpg"
        width="150"
      />
      <p
        className="product_price"
      >
        Choclates
      </p>
    </div>
    <div
      className="product_wrapper"
      id={11}
      onClick={[Function]}
    >
      <img
        alt="category"
        className="category_image"
        height="150"
        src="https://storcpdkenticomedia.blob.core.windows.net/media/recipemanagementsystem/media/recipe-media-files/recipes/retail/x17/16714-birthday-cake-600x600.jpg?ext=.jpg"
        width="150"
      />
      <p
        className="product_price"
      >
        Taffy
      </p>
    </div>
    <div
      className="product_wrapper"
      id={122}
      onClick={[Function]}
    >
      <img
        alt="category"
        className="category_image"
        height="150"
        src="https://storcpdkenticomedia.blob.core.windows.net/media/recipemanagementsystem/media/recipe-media-files/recipes/retail/x17/16714-birthday-cake-600x600.jpg?ext=.jpg"
        width="150"
      />
      <p
        className="product_price"
      >
        Candy Bars
      </p>
    </div>
    <div
      className="product_wrapper"
      id={21}
      onClick={[Function]}
    >
      <img
        alt="category"
        className="category_image"
        height="150"
        src="https://storcpdkenticomedia.blob.core.windows.net/media/recipemanagementsystem/media/recipe-media-files/recipes/retail/x17/16714-birthday-cake-600x600.jpg?ext=.jpg"
        width="150"
      />
      <p
        className="product_price"
      >
        Lollipops
      </p>
    </div>
    <div
      className="product_wrapper"
      id={13}
      onClick={[Function]}
    >
      <img
        alt="category"
        className="category_image"
        height="150"
        src="https://storcpdkenticomedia.blob.core.windows.net/media/recipemanagementsystem/media/recipe-media-files/recipes/retail/x17/16714-birthday-cake-600x600.jpg?ext=.jpg"
        width="150"
      />
      <p
        className="product_price"
      >
        Chewing Gum
      </p>
    </div>
    <div
      className="product_wrapper"
      id={14}
      onClick={[Function]}
    >
      <img
        alt="category"
        className="category_image"
        height="150"
        src="https://storcpdkenticomedia.blob.core.windows.net/media/recipemanagementsystem/media/recipe-media-files/recipes/retail/x17/16714-birthday-cake-600x600.jpg?ext=.jpg"
        width="150"
      />
      <p
        className="product_price"
      >
        Hard Candy
      </p>
    </div>
    <div
      className="product_wrapper"
      id={15}
      onClick={[Function]}
    >
      <img
        alt="category"
        className="category_image"
        height="150"
        src="https://storcpdkenticomedia.blob.core.windows.net/media/recipemanagementsystem/media/recipe-media-files/recipes/retail/x17/16714-birthday-cake-600x600.jpg?ext=.jpg"
        width="150"
      />
      <p
        className="product_price"
      >
        Cotton Candy
      </p>
    </div>
    <div
      className="product_wrapper"
      id={12}
      onClick={[Function]}
    >
      <img
        alt="category"
        className="category_image"
        height="150"
        src="https://storcpdkenticomedia.blob.core.windows.net/media/recipemanagementsystem/media/recipe-media-files/recipes/retail/x17/16714-birthday-cake-600x600.jpg?ext=.jpg"
        width="150"
      />
      <p
        className="product_price"
      >
        Mints
      </p>
    </div>
    <div
      className="product_wrapper"
      id={114}
      onClick={[Function]}
    >
      <img
        alt="category"
        className="category_image"
        height="150"
        src="https://storcpdkenticomedia.blob.core.windows.net/media/recipemanagementsystem/media/recipe-media-files/recipes/retail/x17/16714-birthday-cake-600x600.jpg?ext=.jpg"
        width="150"
      />
      <p
        className="product_price"
      >
        Gems
      </p>
    </div>
  </div>
  <footer
    className="footer"
  >
    <div>
      © 2022 MyStore. All Rights Reserved.
    </div>
  </footer>
</div>
`;
